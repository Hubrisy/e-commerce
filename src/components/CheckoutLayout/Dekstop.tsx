import clsx from 'clsx';
import Link from 'next/link';

import type { CheckLayoutTypes } from '.';

export const DekstopCheckoutLayout: React.FC<CheckLayoutTypes> = ({
  pathname,
  steps,
}) => (
  <div className="flex justify-between">
    {steps.map(item => (
      <div key={item.step} className="flex items-center">
        <Link href={item.path}>
          <div
            className={clsx(
              'flex items-center',
              pathname === item.path ? 'opacity-100' : 'opacity-50',
            )}
          >
            <div>{item.img}</div>
            <div className="ml-2">
              <div className="text-[14px] leading-4">{item.step}</div>
              <div>{item.title}</div>
            </div>
          </div>
        </Link>
      </div>
    ))}
  </div>
);
