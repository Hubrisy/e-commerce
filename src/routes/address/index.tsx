import { AddressBlock } from './AddressBlock';

import { CheckoutLayout } from '@/components/xcheckoutLayout';

const Address = () => (
  <CheckoutLayout>
    <div className="mt-12 mb-3xlarge max-w-[340px] m-auto">
      <div className="text-2xl font-semibold">Address form:</div>
      <AddressBlock />
    </div>
  </CheckoutLayout>
);

export default Address;
