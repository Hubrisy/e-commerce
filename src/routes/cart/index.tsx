import { CartOrderSummary } from './CartOrderSummary';
import { CartProducts } from './CartProducts';

import { useCartContext } from '@/context/Cart';
import { useMountContext } from '@/context/Mount';

const Cart = () => {
  const { cart } = useCartContext();
  const { isMounted } = useMountContext();

  return (
    <div className="mt-[128px] mb-3xlarge max-w-[340px] m-auto md:mt-[72px] lg:max-w-[80%] xl:max-w-[1100px]">
      {isMounted && cart.length > 0 ? (
        <div className="lg:flex justify-between">
          <div className="lg:min-w-[400px] xl:min-w-[500px]">
            <div className="text-2xl font-semibold">Shopping cart:</div>
            <CartProducts />
          </div>
          <CartOrderSummary />
        </div>
      ) : (
        <div className="text-center text-2xl">Cart is empty</div>
      )}
    </div>
  );
};

export default Cart;
