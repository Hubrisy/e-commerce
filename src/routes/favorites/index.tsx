import { ProductItem } from '@/components/productItem';
import { useFavorites } from '@/hooks/use-favorites';
import { useProducts } from '@/hooks/use-products';

const Favorites = () => {
  const { favorites } = useFavorites();
  const { products } = useProducts();

  const arrayOfFavorites = products?.filter(item =>
    favorites.includes(item.id),
  );

  return (
    <div className="max-w-[1100px] mb-3xlarge m-auto">
      <div className="mt-[128px] max-w-[340px] m-auto sm:m-0 sm:ml-10 md:mt-[72px] xl:ml-0">
        <div className="text-2xl font-semibold">Your favorites:</div>
        <div className="grid grid-cols-2 gap-2 mt-4">
          {arrayOfFavorites?.map(item => (
            <div key={item.id}>
              <ProductItem product={item} />
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Favorites;
