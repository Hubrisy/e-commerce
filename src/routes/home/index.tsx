import { BrowseByCategory } from './sections/browseByCategory';
import { MainScreen } from './sections/MainScreen';
import { ProductSection } from './sections/productSection';

import { FilterContextProvider } from '@/context/Filter';
import Footer from '@/sections/footer';
import Header from '@/sections/header';

const Home = () => (
  <FilterContextProvider>
    <div>
      <Header />
      <MainScreen />
      <BrowseByCategory />
      <ProductSection />
      <Footer />
    </div>
  </FilterContextProvider>
);

export default Home;
