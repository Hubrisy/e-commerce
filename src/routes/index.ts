export enum Routes {
  home = '/',
  about = '/about',
  contacts = '/contacts',
  blog = '/blog',
}
