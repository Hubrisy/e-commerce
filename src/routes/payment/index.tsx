import { useEffect } from 'react';
import { useRouter } from 'next/router';

import { Routes } from '..';

import { PaymentForm } from './PaymentForm';

import { Loader } from '@/components/Loader';
import { CheckoutLayout } from '@/components/xcheckoutLayout';
import { useAppStateContext } from '@/context/AppState';
import { useMountContext } from '@/context/Mount';

const Payment = () => {
  const { user } = useAppStateContext();
  const { isMounted } = useMountContext();
  const router = useRouter();

  const isUserValid = Object.values(user).every(value => value.trim() !== '');

  useEffect(() => {
    if (!isUserValid) {
      router.push(Routes.home);
    }
  }, [isUserValid]);

  if (!isMounted || !isUserValid) {
    return (
      <div className="mt-[128px]">
        <Loader />
      </div>
    );
  }

  return (
    <CheckoutLayout>
      <div className="mt-12 mb-3xlarge xl:min-w-[500px]">
        <div className="text-2xl font-semibold">Payment</div>
        <PaymentForm />
      </div>
    </CheckoutLayout>
  );
};

export default Payment;
