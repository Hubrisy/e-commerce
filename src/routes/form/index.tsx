import { FormBlock } from './FormBlock';

import CheckoutLayout from '@/components/CheckoutLayout';
import Layout from '@/components/Layout';

const Form = () => (
  <div className="mt-12 mb-3xlarge max-w-[340px] m-auto">
    <div className="text-2xl font-semibold">Order form:</div>
    <FormBlock />
  </div>
);

Form.getLayout = function getLayout(page: React.ReactNode) {
  return (
    <Layout>
      <CheckoutLayout>{page}</CheckoutLayout>
    </Layout>
  );
};

export default Form;
