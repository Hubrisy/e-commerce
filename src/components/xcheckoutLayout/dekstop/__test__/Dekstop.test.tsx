import React from 'react';
import { render } from '@testing-library/react';

import type { CheckLayoutTypes } from '../..';
import { DekstopCheckoutLayout } from '../Dekstop';

import { Routes } from '@/routes';

jest.mock('next/link', () => {
  const Link = ({
    children,
    href,
  }: {
    children: React.ReactNode;
    href: string;
  }) => <a href={href}>{children}</a>;
  Link.displayName = 'MockLink';

  return Link;
});

const MockLocation = () => <svg data-testid="location-icon" />;
const MockShipping = () => <svg data-testid="shipping-icon" />;
const MockPayment = () => <svg data-testid="payment-icon" />;

describe('DekstopCheckoutLayout', () => {
  const mockSteps: CheckLayoutTypes['steps'] = [
    {
      title: 'Address form',
      step: 'Step 1',
      img: <MockLocation />,
      path: Routes.address,
    },
    {
      title: 'Shipping',
      step: 'Step 2',
      img: <MockShipping />,
      path: Routes.shipping,
    },
    {
      title: 'Payment',
      step: 'Step 3',
      img: <MockPayment />,
      path: Routes.payment,
    },
  ];

  it('should render correctly with first step active', () => {
    const { container } = render(
      <DekstopCheckoutLayout pathname={Routes.address} steps={mockSteps} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with second step active', () => {
    const { container } = render(
      <DekstopCheckoutLayout pathname={Routes.shipping} steps={mockSteps} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with third step active', () => {
    const { container } = render(
      <DekstopCheckoutLayout pathname={Routes.payment} steps={mockSteps} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
