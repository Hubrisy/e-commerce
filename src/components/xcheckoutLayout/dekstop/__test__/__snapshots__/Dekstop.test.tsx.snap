// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DekstopCheckoutLayout should render correctly with first step active 1`] = `
<div
  class="flex justify-between"
>
  <div
    class="flex items-center"
  >
    <a
      href="/address"
    >
      <div
        class="flex items-center opacity-100"
      >
        <div>
          <svg
            data-testid="location-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 1
          </div>
          <div>
            Address form
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/shipping"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="shipping-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 2
          </div>
          <div>
            Shipping
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/payment"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="payment-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 3
          </div>
          <div>
            Payment
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`DekstopCheckoutLayout should render correctly with second step active 1`] = `
<div
  class="flex justify-between"
>
  <div
    class="flex items-center"
  >
    <a
      href="/address"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="location-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 1
          </div>
          <div>
            Address form
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/shipping"
    >
      <div
        class="flex items-center opacity-100"
      >
        <div>
          <svg
            data-testid="shipping-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 2
          </div>
          <div>
            Shipping
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/payment"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="payment-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 3
          </div>
          <div>
            Payment
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`DekstopCheckoutLayout should render correctly with third step active 1`] = `
<div
  class="flex justify-between"
>
  <div
    class="flex items-center"
  >
    <a
      href="/address"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="location-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 1
          </div>
          <div>
            Address form
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/shipping"
    >
      <div
        class="flex items-center opacity-50"
      >
        <div>
          <svg
            data-testid="shipping-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 2
          </div>
          <div>
            Shipping
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    class="flex items-center"
  >
    <a
      href="/payment"
    >
      <div
        class="flex items-center opacity-100"
      >
        <div>
          <svg
            data-testid="payment-icon"
          />
        </div>
        <div
          class="ml-2"
        >
          <div
            class="text-[14px] leading-4"
          >
            Step 3
          </div>
          <div>
            Payment
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
